stream.go

```diff
@@ -140,6 +139,13 @@ func NewClientStream(ctx context.Context, desc *StreamDesc, cc *ClientConn, meth
 		cs.trInfo.tr.LazyLog(&cs.trInfo.firstLine, false)
 		ctx = trace.NewContext(ctx, cs.trInfo.tr)
 	}
+	defer func() {
+		if err != nil {
+			// Need to call cs.finish() if error is returned.
+			// Because cs will not be returned to caller.
+			cs.finish(err)
```

Comment:
`cs` is not even used until way down below after the for loop. Why does it need to be allocated way up here?